Universal FE Randomizer/src/random/gba/randomizer/shuffling/GBACrossGameData.java

```diff
@@ -0,0 +1,424 @@
+			return isFE6 ? FE6Data.CharacterClass.CAVALIER : FE7Data.CharacterClass.CAVALIER_F;
+		} else if (asList(FE8Data.CharacterClass.GREAT_KNIGHT).contains(castedClass)) {
+			return isFE6 ? FE6Data.CharacterClass.GENERAL : FE7Data.CharacterClass.GENERAL;
+		} else if (asList(FE8Data.CharacterClass.KNIGHT).contains(castedClass)) {
+			return isFE6 ? FE6Data.CharacterClass.KNIGHT : FE7Data.CharacterClass.KNIGHT;
+		} else if (asList(FE8Data.CharacterClass.MERCENARY).contains(castedClass)) {
+			return isFE6 ? FE6Data.CharacterClass.MERCENARY : FE7Data.CharacterClass.MERCENARY;
+		} else if (asList(FE8Data.CharacterClass.MERCENARY_F).contains(castedClass)) {
+			return isFE6 ? FE6Data.CharacterClass.MERCENARY : FE7Data.CharacterClass.MERCENARY_F;
+		} else if (asList(FE8Data.CharacterClass.MYRMIDON).contains(castedClass)) {
+			return isFE6 ? FE6Data.CharacterClass.MYRMIDON : FE7Data.CharacterClass.MYRMIDON;
+		} else if (asList(FE8Data.CharacterClass.MYRMIDON_F).contains(castedClass)) {
+			return isFE6 ? FE6Data.CharacterClass.MYRMIDON_F : FE7Data.CharacterClass.MYRMIDON_F;
+		} else if (asList(FE8Data.CharacterClass.PUPIL, FE8Data.CharacterClass.MAGE).contains(castedClass)) {
+			return isFE6 ? FE6Data.CharacterClass.MAGE : FE7Data.CharacterClass.MAGE;
+		} else if (asList(FE8Data.CharacterClass.MAGE_F).contains(castedClass)) {
+			return isFE6 ? FE6Data.CharacterClass.MAGE_F : FE7Data.CharacterClass.MAGE_F;
+		} else if (asList(FE8Data.CharacterClass.PRIEST, FE8Data.CharacterClass.MONK).contains(castedClass)) {
+			return isFE6 ? FE6Data.CharacterClass.PRIEST : FE7Data.CharacterClass.MONK;
+		} else if (asList(FE8Data.CharacterClass.TROUBADOUR).contains(castedClass)) {
+			return isFE6 ? FE6Data.CharacterClass.TROUBADOUR : FE7Data.CharacterClass.TROUBADOUR;
+		} else if (asList(FE8Data.CharacterClass.SAGE).contains(castedClass)) {
+			return isFE6 ? FE6Data.CharacterClass.SAGE : FE7Data.CharacterClass.SAGE;
+		} else if (asList(FE8Data.CharacterClass.SNIPER).contains(castedClass)) {
+			return isFE6 ? FE6Data.CharacterClass.SNIPER : FE7Data.CharacterClass.SNIPER;
+		} else if (asList(FE8Data.CharacterClass.ARCHER_F).contains(castedClass)) {
+			return isFE6 ? FE6Data.CharacterClass.ARCHER_F : FE7Data.CharacterClass.ARCHER_F;
+		} else if (asList(FE8Data.CharacterClass.SHAMAN).contains(castedClass)) {
+			return isFE6 ? FE6Data.CharacterClass.SHAMAN : FE7Data.CharacterClass.SHAMAN;
+		} else if (asList(FE8Data.CharacterClass.CLERIC).contains(castedClass)) {
+			return isFE6 ? FE6Data.CharacterClass.CLERIC : FE7Data.CharacterClass.CLERIC;
+		} else if (asList(FE8Data.CharacterClass.WYVERN_RIDER).contains(castedClass)) {
+			return isFE6 ? FE6Data.CharacterClass.WYVERN_RIDER : FE7Data.CharacterClass.WYVERNKNIGHT;
+		} else if (asList(FE8Data.CharacterClass.ROGUE).contains(castedClass)) {
+			return isFE6 ? FE6Data.CharacterClass.NONE : FE7Data.CharacterClass.ASSASSIN;
+		}
```

Comment:
I usually define these as static HashMaps, which read a bit nicer and can be formatted and can avoid a wall of if-else-ifs.

Reply:
That is certainly possible might be a bit annoying to replace 3 if elses with 6 hashmaps (if f.e. Map<FE6Data.CharacterClass, FE7Data.CharacterClass> 

Might be a bit nicer in that case to only have 3 maps (f.e. Map<FE6Data.CharacterClass, Object[]> where the first object in the array is FE7, and the second FE8, then it can still use Ternary and just needs a cast, which is fine since it's compile time fixed)

Reply:
I was thinking of `Map<FE6Data.CharacterClass, List<GBAFEClass>>`, or even `Map<FE6Data.CharacterClass, Map<GameType, GBAFEClass>>`.

<img width="953" alt="Screenshot 2023-03-26 at 11 55 54 AM" src="https://user-images.githubusercontent.com/1689159/227797792-83b56904-c6ba-4565-b812-4523bef3bd38.png">

Reply:
right, I forgot about the GBAFEClass interface, I wonder if FE6Data.CharacterClass.Soldier and FE7Data.Character.Soldier have the same hashcode, or if we could just put them all in one map of <ClassData, Map<GameType, ClassData>>

Reply:
> I was thinking of `Map<FE6Data.CharacterClass, List<GBAFEClass>>`, or even `Map<FE6Data.CharacterClass, Map<GameType, GBAFEClass>>`.
> 
> <img alt="Screenshot 2023-03-26 at 11 55 54 AM" width="953" src="https://user-images.githubusercontent.com/1689159/227797792-83b56904-c6ba-4565-b812-4523bef3bd38.png">
> 
> EDIT: Better yet: <img alt="Screenshot 2023-03-26 at 11 59 03 AM" width="719" src="https://user-images.githubusercontent.com/1689159/227797938-44c3b750-497b-400d-b5ce-6b3bf6f0d712.png">

The second one doesn't actually work as Map.of() can be used with Maximum of 10 entries (as it's just overloads)

![2023-03-26 22_19_54-Yune_Trunk - Universal FE Randomizer_src_random_gba_randomizer_shuffling_GBACros](https://user-images.githubusercontent.com/41942612/227802419-ff12fd55-f03e-403b-a140-2fe2c7f652c2.png)

Doing it like this might not be that terrible

Reply:
Yeah, that seems fine. I'm actually an iOS developer by trade, so the limitations of Java compared to Swift always throw me when I come back to work on this project 🙃

Reply:
I see, makes sense, I more have the opposite issue, where at work I am stuck on Java 11 due to using Java Enterprise, so I just don't know some of the newer things that are available here